web/src/dialog_widget.ts

```diff
@@ -156,26 +190,26 @@ export function launch(conf) {
 }
 
 export function submit_api_request(
-    request_method,
-    url,
+    request_method: AjaxRequest,
+    url: string,
     data = {},
     {
         failure_msg_html = $t_html({defaultMessage: "Failed"}),
         success_continuation,
         error_continuation,
-    } = {},
-) {
+    }: RequestOpts = {},
+): void {
     show_dialog_spinner();
     request_method({
         url,
         data,
-        success(response_data) {
+        success(response_data?: string) {
```

Comment:
`response_data` is always provided and should not be marked optional. It is not a `string`.